### PR TITLE
Multi contract keystore

### PR DESCRIPTION
**Pull request by gtsonevv**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md).
- [x] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [x] **If this is a code change**: I have written unit tests.
- [x] **If this changes code in a published package**: I have run `pnpm changeset` to create a `changeset` JSON document appropriate for this change.
- [x] **If this is a new API or substantial change**: the PR has an accompanying issue (closes [#1155](https://github.com/near/near-api-js/issues/1153)) and the maintainers have approved on my working plan.

## Test Plan

Added unit tests.

## Related issues/PRs
https://github.com/near/near-api-js/pull/1163

### CODE REVIEW

File: packages/keystores-browser/src/multi_contract_browser_local_storage_key_store.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    private prefix: string;
+
+    /**
+     * @param localStorage defaults to window.localStorage
+     * @param prefix defaults to `near-api-js:keystore:`
+     */
+    constructor(localStorage: any = window.localStorage, prefix = LOCAL_STORAGE_KEY_PREFIX) {
+        super();
+        this.localStorage = localStorage;
+        this.prefix = prefix;
```
Comment (andy-haynes) [inline packages/keystores-browser/src/multi_contract_browser_local_storage_key_store.ts:41]: Can we please make this `this.prefix = prefix || LOCAL_STORAGE_PREFIX;`?

In the current implementation, an empty string prefix means `clear()` would delete all localstorage entries.

**Review by andy-haynes — CHANGES_REQUESTED** [review]
Just one CYA otherwise looking good.

Is it possible to retain the commit history so that Amir remains author on the parts he wrote? I can try to merge the original PR into a feature branch that you could rebase off of if that's easier.

File: packages/keystores-browser/src/multi_contract_browser_local_storage_key_store.ts (hunk 1 above)
Comment (gtsonevv) [reply packages/keystores-browser/src/multi_contract_browser_local_storage_key_store.ts]: Let's merge the original PR into a feature branch.

Comment (andy-haynes) [reply packages/keystores-browser/src/multi_contract_browser_local_storage_key_store.ts]: @gtsonevv go ahead and rebase your changes on top of #1370 and we can merge that once your changes are in.

**Comment by andy-haynes** [comment]
Moved to #1370

**Pull request closed without merging**